core/src/main/java/hudson/model/TopLevelItemDescriptor.java

```diff
@@ -114,6 +127,64 @@ public String getDisplayName() {
     }
 
     /**
+     * A description of this kind of item type. This description can contain HTML code but it is recommend to use text plain
+     * in order to avoid how it should be represented.
+     *
+     * @return A string, an empty string by default.
+     *
+     * @since TODO
+     */
+    @Nonnull
+    public String getDescription() {
+        try {
+            WebApp webapp = WebApp.getCurrent();
+            MetaClass meta = webapp.getMetaClass(this);
+            Script s = meta.loadTearOff(JellyClassTearOff.class).findScript("newInstanceDetail");
+            if (s == null) {
+                return "";
+            }
+            DefaultScriptInvoker dsi = new DefaultScriptInvoker();
+            StringWriter sw = new StringWriter();
+            XMLOutput xml = dsi.createXMLOutput(sw, true);
+            dsi.invokeScript(Stapler.getCurrentRequest(), Stapler.getCurrentResponse(), s, this, xml);
+            return sw.toString();
+        } catch (Exception e) {
+            return "";
+        }
+    }
+
+    /**
+     * Used to categorize this kind of item type. @see {@link ItemCategory}
+     *
+     * @return A string with the category identifier, {@link ItemCategory.UncategorizedCategory#getId()} by default.
+     *
+     * @since TODO
+     */
+    @Nonnull
+    public String getCategoryId() {
+        return new ItemCategory.UncategorizedCategory().getId();
+    }
+
+    /**
+     * @since TODO
+     */
+    @CheckForNull
+    public String getIconFilePathPattern() {
+        return null;
+    }
+
+    /**
+     * @since TODO
+     */
+    @CheckForNull
+    public String getIconFilePath(String size) {
+        if (getIconFilePathPattern().isEmpty()) {
```

Comment:
@jglick In this PR you can find an example where it can be helpful returning a file path pattern instead of a specific file path. As consumer of the web service provides here, I want to know how I can get the icon and then, decide which size I need.